### ISSUE TITLE
new bugs regarding anonymous unions

### ISSUE THREAD

**Issue by TechInterMezzo** [xoofx/CppAst.NET#66]
Following my last issue i found another problem. (Thanks for that fix btw.)

```
struct VECTOR
{
  union
  {
    float fX;
    float fRho;
    float fR;
    float fH;
  };

  union
  {
    float fY;
    float fTheta;
    float fG;
    float fS;
  };

  union
  {
    float fZ;
    float fPhi;
    float fB;
    float fL;
    float fV;
  };
};
```

There are 3 fields in the CppClass object of the VECTOR struct, like expected. One for every anonymous union.
But only the first union gets recognized. All 3 fields have this union as type. The Classes member also only lists this one union, not the following two.

**Comment by TechInterMezzo** [comment]
And one additional "problem".
Every field representing an anonymous union has an Offset of 0. I would expect the second and third field to have an offset of respectively 4 and 8.

**Comment by TechInterMezzo** [comment]
And one more bug to report: even if a union is not anonymous an additional field gets listed.

```
struct STRUCTURE
{
  unsigned long long u64GUID;
  unsigned long long u64OwnerGUID;
  unsigned int u32Flags;
  unsigned int u32RefCount;

  union
  {
    void* stLinkListNode;
    void* stTreeNode;
  } stStorage;

};
```

Although the union has a field named "stStorage" an unnamed field gets listed right before "stStorage".

**Comment by xoofx** [comment]
This should be fixed by commit d0e90c7. A bit annoying to fix, don't like the workaround. (the visitor pattern I'm relying on is not great for this, might have to use at some point the list declarations that have been added to ClangSharp)
